core/src/main/java/io/grpc/transport/AbstractClientStream.java

```diff
@@ -98,7 +98,7 @@ protected void inboundTransportError(Status errorStatus) {
     }
     // For transport errors we immediately report status to the application layer
     // and do not wait for additional payloads.
-    transportReportStatus(errorStatus, false, new Metadata.Trailers());
+    transportReportStatus(errorStatus, true, new Metadata.Trailers());
```

Comment:
The comment isn't really wrong, but is misleading. We should let the error queue behind any _previously_ received data. The only case where it makes a lot of sense to me to jump the queue (and use ```true```) is for cancellation.